### PR TITLE
Raise 400 on batch unknown attributes

### LINKED ISSUE #1163: Batch requests accept additional attributes
We should probably forbid them.

```
> echo '{"requests": [], "foo": {}}' | http post http://localhost:8888/v1/batch 

HTTP/1.1 200 OK
Access-Control-Expose-Headers: Retry-After, Alert, Content-Length, Backoff
Content-Length: 16
Content-Type: application/json
Date: Thu, 16 Mar 2017 23:01:01 GMT
Server: waitress
X-Content-Type-Options: nosniff

{
    "responses": []
}
```

Edit: I just notice we currently test for batch ignoring additional attributes. Is this a feature?

### PR DESCRIPTION
**Pull request by gabisurita**
Fixes #1163

- [x] Add tests.
- [x] Add a changelog entry.
- [x] Add an API changelog entry [in the docs](https://github.com/Kinto/kinto/blob/master/docs/api/index.rst)

### CODE REVIEW

**Comment by Natim** [comment]
Thanks

File: tests/core/test_views_batch.py [hunk 1]
```diff
@@ -247,12 +250,10 @@ def test_defaults_must_be_a_request_schema_if_specified(self):
         batch_payload = {'requests': [request], 'defaults': defaults}
         self.assertInvalid(batch_payload)
 
-    def test_unknown_defaults_are_ignored_silently(self):
+    def test_raise_invalid_on_default_unkown_attributes(self):
```
Comment (glasserc) [inline tests/core/test_views_batch.py]: `unknown` :)

**Comment by Natim** [comment]
This seems to break kinto-signer: https://travis-ci.org/Kinto/kinto-signer/jobs/212165621

**Comment by gabisurita** [comment]
This breaks the current release of kinto-http.py, but it has been fixed on https://github.com/Kinto/kinto-http.py/pull/122

**Comment by Natim** [comment]
So we should release kinto-http.py :)

**Comment by gabisurita** [comment]
> So we should release kinto-http.py :)

We definitely should.

**Pull request merged**